nilearn/decomposition/_base.py

```diff
@@ -198,9 +200,17 @@ def _mask_and_reduce(
     subject_n_samples = [subject_data.shape[0] for subject_data in data_list]
 
     n_samples = np.sum(subject_n_samples)
-    n_voxels = int(np.sum(safe_get_data(masker.mask_img_)))
+    # n_features is the number of True vertices in the mask if it is a surface
+    if isinstance(masker, SurfaceMasker):
+        n_features = (
+            masker.mask_img_.data.parts["left"].sum()
+            + masker.mask_img_.data.parts["right"].sum()
+        )
```

Comment:
```suggestion
        n_features = self.output_dimension_
```
if the masker is fitted you can do this I think

Reply:
and if it's not fitted, I would just fit it